src/Sections/LoginForm.tsx

```diff
@@ -1,37 +1,57 @@
-import Username from "../Components/Username.tsx";
 import React, { useState } from "react";
-import Password from "../Components/Password.tsx";
+import { validatePassword, validatePhoneOrEmail } from "../utils.ts";
 
 interface LoginFormProps {
   submit: (userData: { username: string; password: string }) => void;
 }
 
-function LoginForm({ submit }: LoginFormProps) {
-  const [usernameValid, setUsernameValid] = useState(false);
-  const [passwordValid, setPasswordValid] = useState(false);
-  const [username, setUsername] = useState('');
-  const [password, setPassword] = useState('');
+const LoginForm: React.FC<LoginFormProps> = ({ submit }) => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const isPasswordValid = validatePassword(password);
+  const isUsernameValid = validatePhoneOrEmail(username);
```

Comment:
These aren't computationally expensive, so we can just call the validate functions, even though they are called every time the user types something.

If they were expensive, how would you make this more efficient?